cocos/asset/asset-manager/downloader.ts

```diff
@@ -129,7 +129,7 @@ const downloadBundle = (nameOrUrl: string, options: Record<string, any>, onCompl
     let out: IConfigOption | null = null;
     let error: Error | null = null;
     downloadJson(config, options, (err, response) => {
-        error = err;
+        error = err || error;
```

Comment:
I'm not sure is it necessary to keep the previous error result for each callback @holycanvas

Reply:
Sorry, I tried merging the develop branch, but it added many changed files. If you think my PR is okay. You can take a look at this re-open PR: https://github.com/cocos/cocos-engine/pull/14937